src/hooks/useSections.ts

```diff
@@ -41,19 +48,20 @@ export default function useSections(
   // Fetch Recommendations Results
   const recommendationsResults = useFetchRecommendationPod(cioClient, recommendationsSections);
 
-  const sectionResults = { ...autocompleteResults, ...recommendationsResults };
-
-  const activeSectionsWithData = getActiveSectionsWithData(activeSections, sectionResults);
-
+  // Add to active sections the results data and refs when autocomplete results or recommendation results fetched
   useEffect(() => {
-    setActiveSections(zeroStateActiveSections ? zeroStateSections : sections);
-  }, [query, sections, zeroStateSections, zeroStateActiveSections]);
+    const sectionsResults = { ...autocompleteResults, ...recommendationsResults };
+    setActiveSectionsWithData(
```

Comment:
ActiveSectionsWithData is reactive and get updated based on what get returned form fetchResults